src/arrayMethodSort.js

```diff
@@ -4,8 +4,33 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const array = this;
+
+    const defaultCompare = (a, b) => {
+      const aStr = String(a);
+      const bStr = String(b);
+
+      if (aStr < bStr) {
+        return -1;
+      } else if (aStr > bStr) {
+        return 1;
+      } else {
+        return 0;
+      }
+    };
+
+    const comparator = compareFunction || defaultCompare;
+
+    for (let i = 0; i < array.length - 1; i++) {
```

Comment:
The sorting algorithm implemented here is a bubble sort, which has a time complexity of O(n^2). This is inefficient for large arrays. Consider using a more efficient sorting algorithm like quicksort or mergesort.